internal/pkg/template/template.go

```diff
@@ -48,42 +49,97 @@ type dependencies interface {
 }
 
 type Template struct {
-	dependencies
-	fs           filesystem.Fs
-	fileLoader   filesystem.FileLoader
-	manifest     *Manifest
-	inputs       *Inputs
-	jsonNetCtx   *jsonnet.Context
-	replacements replacekeys.Keys
+	fs       filesystem.Fs
+	srcDir   filesystem.Fs
+	testsDir filesystem.Fs
+	readme   string
+	inputs   *Inputs
 }
 
-func New(fs filesystem.Fs, manifest *Manifest, inputs *Inputs, replacements replacekeys.Keys, d dependencies) *Template {
-	return &Template{
-		dependencies: d,
-		fs:           fs,
-		fileLoader:   fs.FileLoader(),
-		manifest:     manifest,
-		inputs:       inputs,
-		replacements: replacements,
+func New(fs filesystem.Fs, inputs *Inputs) (*Template, error) {
+	// Src dir
+	srcDir, err := fs.SubDirFs(SrcDirectory)
+	if err != nil {
+		return nil, err
 	}
+
+	return &Template{fs: fs, srcDir: srcDir, inputs: inputs}, nil
+}
+
+func (t *Template) ObjectsRoot() filesystem.Fs {
+	return t.srcDir
 }
 
 func (t *Template) Fs() filesystem.Fs {
 	return t.fs
 }
 
-func (t *Template) Manifest() manifest.Manifest {
-	return t.manifest
+func (t *Template) SrcDir() filesystem.Fs {
+	return t.srcDir
+}
+
+func (t *Template) TestsDir() (filesystem.Fs, error) {
+	if t.testsDir == nil {
+		if !t.fs.IsDir(TestsDirectory) {
+			return nil, fmt.Errorf(`directory "%s" not found`, TestsDirectory)
+		}
+		testDir, err := t.fs.SubDirFs(TestsDirectory)
+		if err == nil {
+			t.testsDir = testDir
+		} else {
+			return nil, err
+		}
+	}
+
+	return t.testsDir, nil
+}
+
+func (t *Template) Readme() string {
+	return t.readme
 }
 
 func (t *Template) Inputs() *Inputs {
 	return t.inputs
 }
 
-func (t *Template) Ctx() context.Context {
+func (t *Template) ManifestPath() string {
+	return templateManifest.Path()
+}
+
+func (t *Template) ManifestExists() (bool, error) {
+	return t.srcDir.IsFile(t.ManifestPath()), nil
+}
+
+func (t *Template) ToEvaluated(m *Manifest, jsonNetCtx *jsonnet.Context, replacements replacekeys.Keys, d dependencies) *EvaluatedTemplate {
+	return &EvaluatedTemplate{
```

Comment:
Mame uz 2 structury, `Template` obsahuje `Fs, inputs` ... a `EvaluatedTemplate` obsahuje uz nacitany manifest, ktory sa previedol z `jsonnet -> json`.

Reply:
Vypadá to ok, ale přemýšlím nad tím `evaluated`. 😃  Zní mi to spíš jako bysme ohodnocovali jakou má ta šablona kvalitu (viz. třeba https://www.merriam-webster.com/dictionary/evaluate). Možná by bylo lepší nějaký `processed` (rendered, compiled nebo tak něco)?

Reply:
ok :) ono je to sablona na pol-ceste, ma nacitany manifest, definovane premenne, ale este nema nacitane objekty.

rozmyslam, co tam dat ...

Reply:
hm 🤔  tak co `interpreted`?

Reply:
(koukal jsem do dokumentace jssonnetu jak tomu převodu vlastně říkají 😃)